analysis/analysis.py

```diff
@@ -43,6 +43,47 @@ def compute_total_view_time(student_data):
     return total_view_time
 
 
+def compile_response(student_data, question):
```

Comment:
The purpose of this function wasn't quite clear to me from the docstring -- after examination, I see that it's compiling responses _for a given question_. Maybe a better name would be something like ```get_responses_for_question```?